semgrep-core/src/core/Xpattern.ml

```diff
@@ -21,15 +21,30 @@
 (* Types *)
 (*****************************************************************************)
 
-type regexp = Regexp_engine.t [@@deriving show, eq]
+type compiled_regexp = Regexp_engine.t [@@deriving show, eq]
+type regexp_string = string [@@deriving show, eq]
```

Comment:
maybe a comment here like '(* see the NOTE below on Regexp for the need to have this type *)'